mm/2s2h/Rando/ActorBehavior/TriforceHunt.cpp

```diff
@@ -0,0 +1,43 @@
+#include "ActorBehavior.h"
+#include <libultraship/libultraship.h>
+
+extern "C" {
+#include "variables.h"
+#include "functions.h"
+}
+
+bool isGameplayPaused() {
+    return (Player_InBlockingCsMode(gPlayState, GET_PLAYER(gPlayState)) || gPlayState->pauseCtx.state != 0 ||
+            gPlayState->msgCtx.msgMode != 0)
+               ? true
+               : false;
+}
+
+bool creditsWarpActive = false;
+
+void Rando::ActorBehavior::InitTriforceHuntBehavior() {
+    bool shouldRegister = IS_RANDO && RANDO_SAVE_OPTIONS[RO_SHUFFLE_TRIFORCE_PIECES] == RO_GENERIC_YES;
+
+    COND_ID_HOOK(OnActorUpdate, ACTOR_PLAYER, shouldRegister, [](Actor* actor) {
+        if (!gPlayState) {
+            return;
+        }
+
+        if (gSaveContext.save.shipSaveInfo.rando.foundTriforcePieces ==
```

Comment:
Shouldn't happen, but just in case:
```suggestion
        if (gSaveContext.save.shipSaveInfo.rando.foundTriforcePieces >=
```